### PR TITLE
Fix the error by adding skip processing when incomplete data is included in the RAW block file

### PR DESCRIPTION
**Pull request by azuchi**
If Core's WriteBlockToDisk ftell fails, only the magic byte and size will be written and the block body will be unwritten data. so parsing such data results in the following error:

```
thread 'bulk_writer' panicked at 'indexing failed: Error(Msg("failed to index \"/home/azuchi/.bitcoin/testnet3/blocks/blk00153.dat\""), State { next_error: Some(Error(Msg("failed to parse block at 46716206..46716810"), State { next_error: Some(ParseFailed("Invalid length")), backtrace: InternalBacktrace { backtrace: Some(stack backtrace:
   0: backtrace::backtrace::trace_unsynchronized::hdf11a055294f6f36 (0x55c4f8f652a6)
```

This pull request has been modified to skip such data.

### CODE REVIEW

**Comment by romanz** [comment]
Many thanks for opening the PR!
Would it be possible to add a test for this case?

File: src/bulk.rs [hunk 1]
```diff
@@ -138,6 +138,19 @@ fn parse_blocks(blob: Vec<u8>, magic: u32) -> Result<Vec<Block>> {
             .chain_err(|| format!("seek {} failed", block_size))?;
         let end = cursor.position() as usize;
 
+        // If Core's WriteBlockToDisk ftell fails, only the magic byte and size will be written
+        // and the block body will be unwritten data. skip that's data.
+        let mut tmp_cursor = Cursor::new(&blob[start..(start + 4)]);
+        match u32::consensus_decode(&mut tmp_cursor){
```
Comment (romanz) [inline src/bulk.rs]: Small nit: please run `cargo fmt` to fix the formatting issues.

**Comment by romanz** [comment]
https://github.com/romanz/electrs/issues/113 may be relevant to this PR.

**Comment by azuchi** [comment]
> Would it be possible to add a test for this case?

ok. I will try it.

**Comment by romanz** [comment]
Many thanks, and sorry for the delayed response!

**Pull request merged**